### DISCUSSION TITLE
Issue with .off [@types/sails.io.js]

### DISCUSSION THREAD

**Discussion by daniele-bolla** [DefinitelyTyped/DefinitelyTyped#58770] [Issues with a @types package]
Versions

node v16.14.0 (npm v8.3.1)
@types/sails.io.js: 1.0.5
Typescript: 4.5.5

#### Explain Issue: 

When passing an handler to `.on` the function type is different from `.off`.

Here the example code
```
import socketIOClient from "socket.io-client";
import sailsIOClient from "sails.io.js";
const apiUrl = "http://localhost/sails";
const io = sailsIOClient(socketIOClient);
io.sails.url = apiUrl;
function jobReceived(resp: any) {
  if(resp.done){
    io.socket.off("custom-event", jobReceived);
  }
}
function onCustomEvent() {
  io.socket.on("custom-event",jobReceived);
}

```
Error for `io.socket.off("custom-event", jobReceived);`:

>  'Argument of type '(resp: any) => void' is not assignable to parameter of type '() => any'

#### Possible Fix or Suggestion:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/sails.io.js/index.d.ts

Is it possible to change line 91: ` off(event: string, cb: () => any): ` 
To acccept the same `cb` as the `on` method; i.e. : `on(event: string, cb: (...args: Array<any>) => any): Socket`.

If I'm wrong, could someone suggest or show me a way to handle this situation?

[a comment by an automated account is not shown]

**Comment by martin-badin** [comment]
@daniele-bolla please review #58859
